x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/azureaistudio/AzureAiStudioServiceSettings.java

```diff
@@ -51,63 +52,60 @@ protected static BaseAzureAiStudioCommonFields fromMap(
             AzureAiStudioService.NAME,
             context
         );
-        AzureAiStudioEndpointType endpointType = extractRequiredEnum(
+        AzureAiStudioDeploymentType deploymentType = extractRequiredEnum(
             map,
-            ENDPOINT_TYPE_FIELD,
+            DEPLOYMENT_TYPE_FIELD,
             ModelConfigurations.SERVICE_SETTINGS,
-            AzureAiStudioEndpointType::fromString,
-            EnumSet.allOf(AzureAiStudioEndpointType.class),
+            AzureAiStudioDeploymentType::fromString,
+            EnumSet.allOf(AzureAiStudioDeploymentType.class),
             validationException
         );
-
-        AzureAiStudioProvider provider = extractRequiredEnum(
+        String model = extractOptionalString(
             map,
-            PROVIDER_FIELD,
+            MODEL_FIELD,
             ModelConfigurations.SERVICE_SETTINGS,
-            AzureAiStudioProvider::fromString,
-            EnumSet.allOf(AzureAiStudioProvider.class),
             validationException
         );
 
-        return new BaseAzureAiStudioCommonFields(target, provider, endpointType, rateLimitSettings);
+        return new BaseAzureAiStudioCommonFields(target, deploymentType, model, rateLimitSettings);
     }
 
     protected AzureAiStudioServiceSettings(StreamInput in) throws IOException {
         this.target = in.readString();
-        this.provider = in.readEnum(AzureAiStudioProvider.class);
-        this.endpointType = in.readEnum(AzureAiStudioEndpointType.class);
+        this.deploymentType = in.readEnum(AzureAiStudioDeploymentType.class);
```

Comment:
Similar to my comment below: I think we need to handle the case, where a node using a transport version previous to the one we're introducing with this PR is part of an ES cluster. So we need to read the "old" fields `provider` and `endpointType`, if an old node sends serialized `AzureAiStudioServiceSettings` to a new node.

Something like:

```
if in.getTransportVersion < current version:
   # read old fields to make progress through the serialized data
   # set new fields to some default values and let other code handle the error case for unspecified deploymentType/model etc.
else:
   # only read the new fields
```